app/models/ability.rb

```diff
@@ -0,0 +1,12 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    return unless user.present?
+
+    can :read, Type, user_id: user.id
```

Comment:
@VanjaJeftic I would expect that all users can read all Types and Attributes. Why are you restricting it by user_id?

Reply:
@basti I am restricting it because a user has to be signed in to see all Types and Attributes. When the user signs in, there is by default a list of Types and Attributes that he/she created. On the button 'Show all Types' click, a logged-in user can see all created Types and Attributes. Should I change the implementation logic?

Reply:
@basti I am restricting it because, if I omit that part (user_id: user.id), the login user won't be able to see by default a list of types that he/she created. 
The Implementation logic: when the user signs in, there is by default a list of Types and Attributes that he/she created. On the button 'Show all Types' click, a logged-in user can see all created Types and Attributes. Should I change the implementation logic?

Reply:
@VanjaJeftic that should probably be reversed. Permissions should have priority when defining and then override those load/auth where necessary. So you should probably have `can :read, Type` and then were you need only particular user’s types you add `@types.where(user_id: current_user)` or similar.

The way it stands now abilities file, which should serve as the source of truth of who can do what, says that User should be able to read only his Types. Then in controller you skip over this rule and fetch all. It’s more natural to have it defined correctly in abilities file and then restrict this further when you only need Types from a single user.

Reply:
@basti, yes I understand now, thank you.